mobile/android/android-device-file-system.ts

```diff
@@ -23,12 +24,35 @@ export class AndroidDeviceFileSystem implements Mobile.IDeviceFileSystem {
 		return this.adb.executeShellCommand(listCommandArgs);
 	}
 
-	public getFile(deviceFilePath: string): IFuture<void> {
-		return Future.fromResult();
+	public getFile(deviceFilePath: string, outputPath?: string): IFuture<void> {
+		return (() => {
+			let stdout = false;
+			if (!outputPath) {
+				let fileName = path.basename(deviceFilePath);
+				outputPath = path.join(shelljs.tempdir(), this.$fs.getUniqueFileName(fileName));
+				stdout = true;
+			}
+			this.adb.executeCommand(["pull", deviceFilePath, outputPath]).wait();
+			if (stdout) {
+				let fileToRead = this.$fs.createReadStream(outputPath);
+				let fileToWrite = process.stdout;
```

Comment:
Instead of using `while`, you can just pipe the code to readable stream to `process.stdout`, for example:
```TypeScript
const future = new Future<void>();
readStream.pipe(process.stdout);
readStream.on("end", () => {
	future.return();
});
readStream.on("error", (err: Error) => {
	future.throw(err);
});

future.wait();
```